### ISSUE TITLE
Rule Change: [no-restricted-imports] Add options to specify the behavior for type only import

### ISSUE THREAD

**Issue by Kingwl** [eslint/eslint#15349]
### What rule do you want to change?

no-restricted-imports

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
import xxx from 'module'
import type xxx from 'module'
import type * as xxx from 'module'
import { type xxx } from 'module'
```

### What does the rule currently do for this code?

import xxx from 'module' // error
import type xxx from 'module' // error
import type * as xxx from 'module' // error
import { type xxx } from 'module' // error

### What will the rule do after it's changed?

import xxx from 'module' // should error
import type xxx from 'module' // should not error
import type * as xxx from 'module' // should not error
import { type xxx } from 'module' // should not error

### Participation

- [X] I am willing to submit a pull request to implement this change.

**Comment by nzakas** [comment]
ESLint doesn’t natively support TypeScript, so we don’t have TypeScript-related features in core rules.
